### PR TITLE
ARROW-2745: [C++] ORC ExternalProject needs to declare dependency on vendored protobuf

### PR DESCRIPTION
**Pull request by xhochy**
We need these dependencies as we otherwise get race conditions in the build.

cc @msarahan @kszucs

### CODE REVIEW

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -1056,20 +1056,29 @@ if (ARROW_ORC)
       CMAKE_ARGS ${ORC_CMAKE_ARGS}
       ${EP_LOG_OPTIONS})
 
-    set (ORC_VENDORED 1)
-  else ()
+    set(ORC_VENDORED 1)
+    add_dependencies(orc_ep lz4_static)
```
Comment (kszucs) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake]: `lz4_static` added twice

**Comment by msarahan** [comment]
ah, sorry for the oversimplification.  I only tested the completely vendored and completely external use cases.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (xhochy) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: good catch, fixed.

**Review by wesm — APPROVED** [review]
+1, thanks @xhochy!

**Comment by xhochy** [comment]
Sames Plasma failure as in other PRs.

**Pull request closed without merging**